build.gradle

```diff
@@ -1,34 +1,45 @@
-buildscript {
-	ext {
-		springBootVersion = '2.0.5.RELEASE'
-	}
-	repositories {
-		mavenCentral()
-	}
-	dependencies {
-		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
-	}
-}
+import java.nio.file.Files
 
-apply plugin: 'java'
-apply plugin: 'eclipse'
-apply plugin: 'org.springframework.boot'
-apply plugin: 'io.spring.dependency-management'
+plugins {
+    id 'java'
+    id 'org.springframework.boot' version '2.0.5.RELEASE'
+    id 'io.spring.dependency-management' version '1.0.5.RELEASE'
+}
 
+sourceCompatibility = '1.8'
+targetCompatibility = '1.8'
 group = 'com.scottlogic.competition_engine'
 version = '0.0.1-SNAPSHOT'
-sourceCompatibility = 1.8
 
 repositories {
-	mavenCentral()
+    mavenCentral()
 }
 
-
 dependencies {
-//	implementation('org.springframework.boot:spring-boot-starter-data-jpa')
-	implementation('org.springframework.boot:spring-boot-starter-websocket')
-//	implementation('org.flywaydb:flyway-core')
-//	runtimeOnly('org.postgresql:postgresql')
-	compileOnly('org.projectlombok:lombok')
-	testImplementation('org.springframework.boot:spring-boot-starter-test')
+    implementation('org.springframework.boot:spring-boot-starter-websocket')
+    compileOnly('org.projectlombok:lombok')
+    testImplementation('org.springframework.boot:spring-boot-starter-test')
 }
+
+task copyRenameJar {
+    def latestFile
+    doFirst {
+        def files = file('./build/libs/').listFiles().sort()
+        def jarFiles = []
+        files.each { File file ->
+            if (file.name.endsWith('.jar')) {
+                jarFiles.add(file)
+            }
+        }
+        latestFile = jarFiles.last()
+    }
+    doLast {
+        copy {
+            from file(latestFile)
+            into '.'
+            rename { "SNAPSHOT.jar" }
+        }
+    }
+}
```

Comment:
There are a couple of problems with this task definition (and they all have one solution):
  - Hard-coded file paths are a bad idea. If, for some reason, the task that builds the jar changes where it puts it in the future, then this will need to be updated too, but it might get missed if an _old_ built jar is still in the old location. In Gradle, you can look up the output path of a particular task.
  - This task doesn't depend on the task that builds the jar ('jar'), so tehre's no guarantee that the copied jar won't be out-of-date.
  - Sorting here relies on the assumption that lexical order is the same as version order. This isn't necessarily true, particularly if the jar's 'base name' gets changed in the futrure.

I believe you can achieve the same result with the following:

```
task('copyRenameJar', type: Copy) {
    from jar
    into projectDir
    rename { 'SNAPSHOT.jar' }
}
```
Let me know if you don't understand how this works, or if I've missed something here. I'm WFH today, but I'll be back in the office tomorrow.

Reply:
On a separate note, maybe we should also add 'SNAPSHOT.jar' to the .gitignore file.

Reply:
Yeah I had some problems getting the type:Copy way working so I've resorted to doing it this way instead.  If you have time tomorrow and could chat about it, that'd be great.

Also, I didn't realise that about the PRs so I'll do that in future.

Reply:
OK cool - I can go through stuff with you tomorrow.